packages/next/src/build/create-compiler-aliases.ts

```diff
@@ -90,6 +93,12 @@ export function createWebpackAliases({
   return {
     '@vercel/og$': 'next/dist/server/og/image-response',
 
+    // Avoid bundling both entrypoints in React 19 when we just need one.
+    // Also avoids bundler warnings in React 18 where react-dom/server.edge doesn't exist.
+    'next/dist/server/ReactDOMServerPages': isReact19
+      ? 'react-dom/server.edge'
+      : 'react-dom/server.browser',
```

Comment:
Ideally this would use the same approach as Turbopack and use 
```suggestion
    'next/dist/server/ReactDOMServerPages': [
      'react-dom/server.edge'
      'react-dom/server.browser'
    ],
```
i.e. fallbacks. But that doesn't seem to work and Webpack still tries to bundle `react-dom/server.edge` leading to 
```
Module not found: Package path ./server.edge is not exported from package
```

when running
`NEXT_TEST_MODE=dev NEXT_TEST_REACT_VERSION="18.3.1" p test test/e2e/app-dir/app-edge/app-edge.test.ts`